core/modules.rs

```diff
@@ -30,10 +30,19 @@ pub trait Loader {
   /// Given an absolute url, load its source code.
   fn load(&mut self, url: &str) -> Box<SourceCodeFuture<Self::Error>>;
 
-  fn enter<R, F: FnMut(&mut Isolate<Self::Behavior>, &mut Modules) -> R>(
-    &mut self,
-    cb: F,
-  ) -> R;
+  fn isolate_and_modules<'a: 'b + 'c, 'b, 'c>(
+    &'a mut self,
+  ) -> (&'b mut Isolate<Self::Behavior>, &'c mut Modules);
+
+  fn isolate<'a: 'b, 'b>(&'a mut self) -> &'b mut Isolate<Self::Behavior> {
+    let (isolate, _) = self.isolate_and_modules();
+    isolate
+  }
+
+  fn modules<'a: 'b, 'b>(&'a mut self) -> &'b mut Modules {
+    let (_, modules) = self.isolate_and_modules();
+    modules
+  }
```

Comment:
Can we remove (`isolate` and `modules`) OR remove `isolate_and_modules` ?

Reply:
We'll deal with this later during integration. Landing now.

Reply:
`isolate_and_modules()` can't be removed because we need to borrow them simultaneously at a certain point.
The other ones could go -- but it makes RecursiveLoad::poll() bigger and uglier.